File: ruby/benchmark.rb
require 'fast_excel'

ROWS = 10000
COLS = 50

def generate_test_data()
	data = []
	0.upto(ROWS - 1) do |row|
		0.upto(COLS - 1) do |col|
			data[row] ||= []
			data[row][col] = row + col
		end
	end
	data
end

def test()
	test_data = generate_test_data()

	begin_timestamp = Time.now().to_f

	workbook = FastExcel.open("output/ruby.xlsx", constant_memory: true)
	sheet = workbook.add_worksheet("Example")
	test_data.each do |row_data|
		sheet.append_row row_data
	end

	workbook.close
	end_timestamp = Time.now().to_f

	puts "Ruby: Writing 10000x50 cells of data takes #{end_timestamp - begin_timestamp} seconds"
end

test()

File: python/benchmark.py
import time

from pyexcelerate import Workbook

ROWS = 100000
COLS = 50

def generate_data():
	data = [[0] * COLS] * ROWS
	for row in range(ROWS):
		for col in range(COLS):
			data[row][col] = row + col
	return data

def test():
	begin_timestamp = time.time()
	wb = Workbook()
	wb.new_sheet("sheet name", data=generate_data())
	wb.save("output/python.xlsx")
	end_timestamp = time.time()

	print("Python: Writing 10000x50 cells of data takes %f seconds" % (end_timestamp - begin_timestamp))


if __name__ == '__main__':
	test()
